src/vmm/src/arch/aarch64/vcpu.rs

```diff
@@ -153,12 +153,25 @@ pub fn get_registers(
 /// Returns all registers ids, including core and system
 pub fn get_all_registers_ids(vcpufd: &VcpuFd) -> Result<Vec<u64>, VcpuError> {
     // Call KVM_GET_REG_LIST to get all registers available to the guest. For ArmV8 there are
-    // less than 500 registers.
-    let mut reg_list = RegList::new(500).map_err(VcpuError::Fam)?;
-    vcpufd
-        .get_reg_list(&mut reg_list)
-        .map_err(VcpuError::GetRegList)?;
-    Ok(reg_list.as_slice().to_vec())
+    // less than 500 registers expected, resize to the reported size when necessary.
+    let mut reg_list = RegList::new(10).map_err(VcpuError::Fam)?;
+
+    match vcpufd.get_reg_list(&mut reg_list) {
+        Ok(_) => Ok(reg_list.as_slice().to_vec()),
+        Err(e) => match e.errno() {
+            libc::E2BIG => {
+                // resize and retry.
+                reg_list = RegList::new(reg_list.as_fam_struct_ref().n.try_into().unwrap())
```

Comment:
Could we replace the `unwrap()` with an error propagation?
As a nit, we can extract the new size into a variable to make it easier for readers to see what we're doing.

Reply:
thanks! added comment.